### PR TITLE
Add serialization for RenderingSettings.

### PR DESCRIPTION
**Pull request by siarheishut**
Add serialization and deserialization for RenderingSettings. 
Add unit-test for it.

### CODE REVIEW

File: tests/rendering_settings_test.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    auto &stop_label_offset = *settings.mutable_stop_label_offset();
+    stop_label_offset.set_x(-5.0);
+    stop_label_offset.set_y(2.0);
+    settings.mutable_underlayer_color()->set_name("red");
+    settings.set_underlayer_width(5.0);
+    auto &color_palette = *settings.mutable_color_palette();
+    *color_palette.Add() = SerializeColor(svg::Rgba{1, 100, 20, 0.1});
+    color_palette.Add()->set_name("purple");
+    *color_palette.Add() = SerializeColor(svg::Rgb{15, 23, 41});
+    *color_palette.Add() = SerializeColor(svg::Rgba{210, 81, 14, 0.94});
```
Comment (foxfrikses) [inline tests/rendering_settings_test.cpp:52]: none color?

File: tests/rendering_settings_test.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    auto &bus_label_offset = *settings.mutable_bus_label_offset();
+    bus_label_offset.set_x(5.0);
+    bus_label_offset.set_y(-7.0);
+    auto &layers = *settings.mutable_layers();
+    *layers.Add() = ::TransportCatalog::MapLayer::kBusLabels;
+    *layers.Add() = ::TransportCatalog::MapLayer::kStopLabels;
+    *layers.Add() = ::TransportCatalog::MapLayer::kStopPoints;
+    *layers.Add() = ::TransportCatalog::MapLayer::kStopPoints;
+    *layers.Add() = ::TransportCatalog::MapLayer::kBusLines;
+    *layers.Add() = ::TransportCatalog::MapLayer::kBusLabels;
```
Comment (foxfrikses) [inline tests/rendering_settings_test.cpp:63]: `    *layers.Add() = static_cast<::TransportCatalog::MapLayer>(1000);`

File: src/request_types.h [hunk 3]
```diff
@@ -31,6 +32,11 @@ enum class MapLayer {
 struct Frame { double width, height, padding; };
 
 struct RenderingSettings {
+  static std::unique_ptr<utils::RenderingSettings> Deserialize(
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/request_types.h:35]: Done.

File: tests/rendering_settings_test.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
+#include "rendering_settings.pb.h"
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply tests/rendering_settings_test.cpp:1]: Done.

File: tests/rendering_settings_test.cpp (hunk 1 above)
Comment (siarheishut) [reply tests/rendering_settings_test.cpp:52]: Done.

File: tests/rendering_settings_test.cpp (hunk 2 above)
Comment (siarheishut) [reply tests/rendering_settings_test.cpp:63]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**